cmd/snap-repair/cmd_list.go

```diff
@@ -0,0 +1,183 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/snapcore/snapd/dirs"
+)
+
+func init() {
+	const (
+		short = "List repairs run on this device"
+		long  = ""
+	)
+
+	if _, err := parser.AddCommand("list", short, long, &cmdList{}); err != nil {
+		panic(err)
+	}
+
+}
+
+type cmdList struct {
+	Verbose bool `long:"verbose"`
+}
+
+func outputIndented(w io.Writer, path string) {
+	f, err := os.Open(path)
+	if err != nil {
+		fmt.Fprintf(w, "  error: %s\n", err)
+		return
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		fmt.Fprintf(w, "  %s\n", scanner.Text())
+	}
+	if scanner.Err() != nil {
+		fmt.Fprintf(w, "  error: %s\n", scanner.Err())
+	}
+
+}
+
+func showRepairOutput(w io.Writer, issuer, seq, rev string) error {
+	basedir := filepath.Join(dirs.SnapRepairRunDir, issuer, seq)
+	dirents, err := ioutil.ReadDir(basedir)
+	if err != nil {
+		return err
+	}
+	for _, dent := range dirents {
+		name := dent.Name()
+		if strings.HasSuffix(name, ".retry") || strings.HasSuffix(name, ".done") || strings.HasSuffix(name, ".skip") {
+			fmt.Fprintf(w, " output:\n")
+			outputIndented(w, filepath.Join(basedir, name))
+		}
+		if strings.HasPrefix(name, "script.") {
+			fmt.Fprintf(w, " script:\n")
+			outputIndented(w, filepath.Join(basedir, name))
+		}
+	}
+
+	return nil
+}
+
+type repairTrace struct {
+	issuer string
+	seq    string
+	rev    string
+	status string
+}
+
+func newRepairTrace(artifactName, issuerName, seqName, status string) repairTrace {
+	t := repairTrace{
+		issuer: issuerName,
+		seq:    seqName,
+		rev:    "?",
+		status: status,
+	}
+	var rev int
+	if _, err := fmt.Sscanf(artifactName, "r%d.", &rev); err == nil {
```

Comment:
Do you want to do something about the error in case it happens?

Reply:
I think that is fine to ignore the error, this is really about finding files that belong to  a given repair. If there is other stuff in the dir we can just ignore it.